onnxruntime/core/providers/cuda/math/gemm.cc

```diff
@@ -38,25 +48,26 @@ template <typename T>
 Status Gemm<T>::ComputeInternal(OpKernelContext* ctx) const {
   typedef typename ToCudaType<T>::MappedType CudaT;
 
-  const auto X = ctx->Input<Tensor>(0);
-  const auto W = ctx->Input<Tensor>(1);
-  const auto B = ctx->Input<Tensor>(2);
-  GemmHelper helper(X->Shape(), trans_A_, W->Shape(), trans_B_, B->Shape());
+  const auto* X = ctx->Input<Tensor>(0);
+  const auto* W = ctx->Input<Tensor>(1);
+  const auto* B = ctx->InputCount() == 3 ? ctx->Input<Tensor>(2) : nullptr;
```

Comment:
>ctx->InputCount() == 3 ? [](start = 18, length = 24)

ctx->Input does a check on input count so it's not really necessary here

Reply:
I see. So I presume it returns a nullptr if the input doesn't exist ?

Reply:
Yes

Reply:
Made the change